src/lib.rs

```diff
@@ -512,3 +515,22 @@ fn ioctl_to_error(ret: i32) -> io::Result<i32> {
         Ok(ret)
     }
 }
+
+// Android doesn't use devtmpfs. Instead device nodes under /dev are
+// created by userspace daemon ueventd. There could be a noticeable delay
+// between LOOP_CTL_GET_FREE issued and loop device created, so we need to
+// wait until it is created and then continue.
+// The timeout (5s) and time quantum (20ms) is picked randomly, it bares no
+// special meaning but they worked fine empirically.
+#[cfg(target_os = "android")]
+fn wait_for_device<P: AsRef<Path>>(device: P) {
+    let start = std::time::Instant::now();
+    let timeout = std::time::Duration::from_secs(5);
+    let quantum = std::time::Duration::from_millis(20);
+    while !device.as_ref().exists() {
+        if start.elapsed() > timeout {
+            break;
+        }
+        std::thread::sleep(quantum);
+    }
```

Comment:
It's very likely that the first `exits` call returns `false`. This results in a at least 5ms delay which can be unacceptable long. 
Setting up [inotify](https://man7.org/linux/man-pages/man7/inotify.7.html) might also take some time but my feeling is, that it can be faster than 5ms.